store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java

```diff
@@ -1686,6 +1690,16 @@ public boolean checkInDiskByCommitOffset(long offsetPy) {
         return offsetPy >= commitLog.getMinOffset();
     }
 
+    /**
+     * The ratio val is estimated by the experiment and experience
+     * so that the result is not high accurate for different business
+     * @return
+     */
+    public boolean checkInColdAreaByCommitOffset(long offsetPy, long maxOffsetPy) {
```

Comment:
IMO, using linux system call mincore() method to check file been cached or not will achieve higher accuracy, just like pcstat command in https://github.com/tobert/pcstat

Reply:
This method mainily helps us distinguish between logical cold data and not. Sometimes it can prevent normal consumers from being cold controlled.

Reply:
Thank you for your review ～